sql/analyzer/optimization_rules.go

```diff
@@ -327,6 +327,11 @@ func simplifyFilters(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope,
 
 				return e, transform.SameTree, nil
 			case *expression.Like:
+				// if the charset is utf8mb3, the last character used in optimization rule does not work
+				charset := getCharsetInUse(ctx, e.Left)
+				if charset == sql.CharacterSet_utf8mb3 {
```

Comment:
This seems very wrong. If we're seeing issues with `utfmb3`, I suspect it's due to `LIKE` just not working properly on anything that isn't `utf8mb4`. This should be `if charset != sql.CharacterSet_utf8mb4` instead.

Reply:
I thought so too, some tests that were using `binary` charset were failing when I had `if charset != sql.CharacterSet_utf8mb4` which seemed weird. I'll double check on them.